com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -236,12 +214,12 @@ private boolean shouldRecreate(CloudServiceExtended service, CloudService existi
             return false;
         }
         ServiceOperation lastOperation = getLastOperation(serviceInstanceEntity);
-        if (lastOperation != null
-            && (lastOperation.getType() == ServiceOperationType.CREATE || lastOperation.getType() == ServiceOperationType.DELETE)
+        if (lastOperation != null && lastOperation.getType() == ServiceOperationType.CREATE
```

Comment:
maybe extract this check like the hasServiceDeleteFailedState(lastOperation)

Reply:
done